src/Method.cpp

```diff
@@ -369,6 +369,12 @@ bool Method::removeBlock(BasicBlock& block, bool overwriteUsages)
     return false;
 }
 
+BasicBlock& Method::createAndInsertNewBlock(BasicBlockList::iterator position, const std::string& labelName)
+{
+    return *basicBlocks.emplace(
+        position, *this, new intermediate::BranchLabel(*findOrCreateLocal(TYPE_LABEL, labelName)));
```

Comment:
You should not search for an existing label here. If the label already exists, then we will have a problem!